### PR TITLE
Streamline typeCheck and add initial test suite

### PR DESCRIPTION
**Pull request by alchemyst**
I've started by making typeCheck easier to change by changing the order and contents of the types variable rather than redoing logic. In the process I've also started a small test suite to make sure I don't mess anything up with the edits.

I am a bit puzzled by what kind of thing will raise ValueError when passed to `str` in Python, so the final tests are empty right now.

### CODE REVIEW

File: OMPython/OMParser/__init__.py [hunk 1]
```diff
@@ -41,36 +41,28 @@
 next_set = []
 next_set.append('')
 
+def bool_from_string(string):
+    if string in {'true', 'True', 'TRUE'}:
```
Comment (bilderbuchi) [inline OMPython/OMParser/__init__.py]: That would be simpler with `if string.lower() == 'true':

File: OMPython/OMParser/__init__.py [hunk 2]
```diff
@@ -41,36 +41,28 @@
...
+        return True
+    elif string in {'false', 'False', 'FALSE'}:
```
Comment (bilderbuchi) [inline OMPython/OMParser/__init__.py]: see above

File: tests/__init__.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+from OMPython import OMParser
```
Comment (bilderbuchi) [inline tests/__init__.py]: why are you importing here and in the test_*.py files? is this a unittest quirk? (I'm mainly familiar with pytest)

Comment (bilderbuchi) [inline tests/__init__.py]: btw, :+1: on including tests in your code!

File: OMPython/OMParser/__init__.py (hunk 1 above)
Comment (alchemyst) [inline OMPython/OMParser/__init__.py]: I also thought about that, but that would actually do something different from the current implementation, which only checks for those particular strings. This change was meant to be a pure refactor.

File: tests/__init__.py (hunk 3 above)
Comment (alchemyst) [inline tests/__init__.py]: I definitely need to import from the `test_*.py` files for the tests to run (using `python -m unittest tests tests.test_OMParser` from the base directory). I'm not very used to working with packages (the current structure seems very baroque to me) but this is what I had to do to get the tests going. I'm open to improvement.

File: OMPython/OMParser/__init__.py (hunk 1 above)
Comment (bilderbuchi) [inline OMPython/OMParser/__init__.py]: OK, right, this makes sense.

File: tests/__init__.py (hunk 3 above)
Comment (bilderbuchi) [inline tests/__init__.py]: Yeah, i was primarily wondering about i.porting it in __init__.py, then not using it.

Comment (alchemyst) [inline tests/__init__.py]: You're right. I've removed it.

**Comment by alchemyst** [comment]
Is there anything else I can change to have this pull request accepted?

**Comment by bilderbuchi** [comment]
we can ping @sjoelund...

**Comment by alchemyst** [comment]
Any news on this in the new year?

**Comment by bilderbuchi** [comment]
I don't know, I'm just a drive-by potential user. @adeas31 apparently has merge rights, as he recently merged another PR, so maybe he can take a look?

**Comment by alchemyst** [comment]
Thanks @hkiel 😀

**Pull request merged**